### PR TITLE
wasitest: add tests to validate the polling of sockets before they are connected

### PR DESCRIPTION
**Pull request by achille-roussel**
Those tests verify edge cases when sockets are polled on before they are connected:
- for stream sockets, no events should trigger
- for datagram sockets, writing is supported as soon as the socket is created so the write event must trigger

### CODE REVIEW

**Review by chriso — APPROVED** [review]

**Review by pelletier — APPROVED** [review]

File: wasitest/socket.go [hunk 1]
```diff
@@ -820,8 +910,10 @@ func testSocketConnectOK(family wasi.ProtocolFamily, typ wasi.SocketType, bind w
 		addr, errno := sys.SockConnect(ctx, sock, bind)
 		assertNotEqual(t, addr, nil)
 		assertEqual(t, addr.Family(), bind.Family())
-		if errno != wasi.ESUCCESS {
+		if typ == wasi.StreamSocket {
 			assertEqual(t, errno, wasi.EINPROGRESS)
+		} else {
+			assertEqual(t, errno, wasi.ESUCCESS)
 		}
```
Comment (chriso) [inline wasitest/socket.go:917]: Is this true? I think connecting to the loopback may succeed immediately?

Comment (achille-roussel) [reply wasitest/socket.go:936]: The test connects to the loopback and it seems to always give EINPROGRESS on both darwin and linux.
